core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -193,7 +193,7 @@ protected function import($strClass, $strKey=null, $blnForce=false)
 			}
 			elseif (\in_array('getInstance', get_class_methods($strClass)))
 			{
-				static::$arrStaticObjects[$strKey] = static::$arrSingletons[$strKey] = $this->arrObjects[$strKey] = \call_user_func(array($strClass, 'getInstance'));
+				static::$arrStaticObjects[$strKey] = static::$arrSingletons[$strClass] = $this->arrObjects[$strKey] = \call_user_func(array($strClass, 'getInstance'));
```

Comment:
I think we should not write to `static::$arrStaticObjects[$strKey]` in this case either.

Reply:
I agree. It should not really matter, though, because the classes are singletons anyway.

Reply:
Ah, it does matter. If we remove it, we have to add a `elseif (isset(static::$arrSingletons[$strClass]))` clause in the `importStatic()` method.

Reply:
Adjusted in 02fd655f61a2902aff5444062a68fdf6242815a7.